modules/ingest-download-service-impl/src/main/java/org/opencastproject/ingestdownloadservice/impl/IngestDownloadServiceImpl.java

```diff
@@ -0,0 +1,348 @@
+
+  }
+
+  @Override
+  protected String process(Job job) throws MediaPackageException, IOException {
+    final List<String> arguments = new ArrayList<>(job.getArguments());
+
+    final MediaPackage mediaPackage = MediaPackageParser.getFromXml(arguments.get(0));
+    final String sourceFlavors = arguments.get(1);
+    final String sourceTags = arguments.get(2);
+    final boolean deleteExternal = Boolean.parseBoolean(arguments.get(3));
+    final boolean tagsAndFlavor = Boolean.parseBoolean(arguments.get(4));
+
+    // building elementSelector with tags and flavors
+    AbstractMediaPackageElementSelector<MediaPackageElement> elementSelector = new SimpleElementSelector();
+    for (String tag : StringUtils.split(sourceTags, ", ")) {
+      elementSelector.addTag(tag);
+    }
+    for (String flavor : StringUtils.split(sourceFlavors, ", ")) {
+      elementSelector.addFlavor(flavor);
+    }
+
+    final String baseUrl = workspace.getBaseUri().toString();
+
+    List<URI> externalUris = new ArrayList<>();
+    for (MediaPackageElement element : elementSelector.select(mediaPackage, tagsAndFlavor)) {
+      if (element.getURI() == null) {
+        continue;
+      }
+
+      if (element.getElementType() == MediaPackageElement.Type.Publication) {
+        logger.debug("Skipping publication {} from media package {}", element.getIdentifier(),
+                     mediaPackage.getIdentifier());
+        continue;
+      }
+
+      if (element.getURI().toString().startsWith(baseUrl)) {
+        logger.info("Skipping already existing element {}", element.getURI());
+        continue;
+      }
+
+      // Download the external URI
+      File file;
+      try {
+        file = workspace.get(element.getURI());
+      } catch (NotFoundException e) {
+        logger.warn("Unable to download the external element {}", element.getURI());
+        continue;
+      }
+
+      // Put to working file repository and rewrite URI on element
+      final URI originalUri = element.getURI();
+      try (InputStream in = new FileInputStream(file)) {
+        final String filename = FilenameUtils.getName(element.getURI().getPath());
+        final URI uri = workspace.put(mediaPackage.getIdentifier().compact(), element.getIdentifier(), filename, in);
+        element.setURI(uri);
+      } finally {
+        try {
+          workspace.delete(originalUri);
+        } catch (Exception e) {
+          logger.warn("Unable to delete ingest-downloaded element {}", element.getURI(), e);
+        }
+      }
+
+      logger.info("Downloaded the external element {}", originalUri);
+
+      // Store original URI for deletion
+      externalUris.add(originalUri);
+    }
+
+    if (!deleteExternal || externalUris.size() == 0)
+      return MediaPackageParser.getAsXml(mediaPackage);
+
+    // Find all external working file repository base Urls
+    logger.debug("Assembling list of external working file repositories");
```

Comment:
Ok, wait a minute.  You're checking for the case where there is more than one WFR in the cluster, and then telling it to remove the file(s) if they exist.  What's the usecase here?  If we're fetching a file from an external host, how can said file already be present in another WFR?

Reply:
Hmm good question. I just copied the old code here. 
Maybe this is possible if we have a cluster without a shared (nfs)medium?
But to have a workspace and a workingfile-repository seems to me, that we have one storage to much... 

If this check is not necessary, i would be happy to remove it.

Reply:
If this was copy and pasted then likely the check originated back in the days when the WFR could be in multiple places.  I don't think it's likely to be useful now, but there might be some weird edge case that would show up later.  Let's leave it for now I guess, at some point someone will rip out the WFR entirely, and they will remove this check at that point.